### PR TITLE
Bugfix/plugin prefix

### PR DESCRIPTION
**Pull request by titouanmathis**
This PR adds prefixes to the classes generated by the grid and the debug outline plugins.

The 1.2.0 release of Tailwind introduced utilities for the grid layout with a `.grid` class, conflicting with our own `.grid` class in the grid plugin. 

In order to avoid similar conflicts in the future, I think we have to add a `prefix` options to all our custom plugins.

The documentations have been updated:
- [Debug outline](https://deploy-preview-18--tailwind-config.netlify.com/plugins/debug-outline.html#debugoutline-prefix)
- [Grid](https://deploy-preview-18--tailwind-config.netlify.com/plugins/grid.html#prefix)

### CODE REVIEW

**Comment by perruche** [comment]
Is the `b-` and `c-` prefixes for `base` and `components` ?  
This will be the standard for all our custom plugins ?

**Comment by titouanmathis** [comment]
Oops, this should not have been closed. Reopening the PR 🙂

**Comment by titouanmathis** [comment]
> Is the `b-` and `c-` prefixes for `base` and `components` ?

Yes that's the proposal:
- `b-` for base 
- `c-` for components
- `u-` for utilities

With the option to disable the prefix in the configuration.

> This will be the standard for all our custom plugins ?

And also yes: we do not know what classes future releases of Tailwind CSS will add, so it should be safer to already prefix our custom classes.

**Comment by lusimeon** [comment]
And what if tailwind creates a utility with a `c-` or `b-` prefix ? Maybe we should set a more specific prefix…

**Comment by antoine4livre** [comment]
I propose to use another separator for our prefixes like `·` (middle point `option`+`maj`+`F`) or the `::` already tested with titou but not quite convincing. Do you have other ideas?

**Comment by antoine4livre** [comment]
@ptt-homme proposed to simply prefix with an underscore or double underscore at the beginning of the class.
Example: `._grid` ou `.__grid`

**Comment by perruche** [comment]
I don't think Tailwind will ever use prefixes like `b-` or `c-`, cause you already have the possibility to use a [custom prefix](https://tailwindcss.com/docs/configuration/#prefix) for all of the tailwind classes. 

Not a fan of using underscores or a special caracter for a prefix, this require pressing multiple keystrokes 🤕

**Comment by ptt-homme** [comment]
I'm not really fan of this kind of prefix `c-` or `-b` which have no real meaning about what they do. Why not using on purpose a more meaningful prefix such as `meta-grid` or `custom-grid` to avoid conflicts and let others components without prefix... This will be shorter for a daily basis work and can be easily fix, if so.
There is no way to remove `.grid` from tailwind initial config? Or even better, do a plugin interface to change which grid system is used on purpose?

**Comment by perruche** [comment]
Currently not possible to remove `.grid` @antoine4livre and @titouanmathis have looked into it.
but this [open issue](https://github.com/tailwindcss/tailwindcss/issues/1364) seems related to our problem.

**Comment by titouanmathis** [comment]
> I'm not really fan of this kind of prefix c- or -b which have no real meaning about what they do. 

I think `c-` is quite easy to understand as we already use such prefixes for states (`is-` and `has-`) and for JS hooks (`js-`) and it follows best practices recommandations ([1](https://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces/#the-namespaces), [2](https://csswizardry.com/2015/08/bemit-taking-the-bem-naming-convention-a-step-further/#namespaces), [3](https://www.smashingmagazine.com/2016/06/battling-bem-extended-edition-common-problems-and-how-to-avoid-them/#2-should-i-be-namespacing)).

The `b-` prefix is not so easy to understand, and we might replace it by a `c-` prefix as the debug outline plugin should actually be a component rather than a base.

To sum up, I see the following advantages for the `c-`prefix:
- it is short
- it follows CSS namespacing best practices / recommandations
- it is a simple rule 
- it ensures consistency in prefix usage

**Comment by titouanmathis** [comment]
I updated the configuration as follow:

- set an empty string as default for the plugins prefixes
- disable the core plugin `display`
- add a copy of the core plugin `display`, but without the `.grid` class

**Comment by ptt-homme** [comment]
🍾💪

Le mar. 11 févr. 2020 à 16:10, Titouan Mathis <notifications@github.com> a
écrit :
